### PR TITLE
Add new AWS module - eks-service

### PR DESCRIPTION
**Pull request by spartan-tainguyen**
## Summary

This module helps configure an 'EKS Service' by creating a correlation set of Kubernetes and AWS resources.

## Types of Changes

- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🕷 Bug fix (non-breaking change which fixes an issue)
- [ ] 👏 Performance optimization (non-breaking change which addresses a performance issue)
- [ ] 🛠 Refactor (non-breaking change which does not change existing behavior or add new functionality)
- [ ] 📗 Library update (non-breaking change that will update one or more libraries to newer versions)
- [ ] 📝 Documentation (non-breaking change that doesn't change code behavior, can skip testing)
- [ ] ✅ Test (non-breaking change related to testing)
- [ ] 🔒 Security awareness (changes that effect permission scope, security scenarios)

## Related Issues

https://www.notion.so/c0x12c/Infra-Rollout-new-Terraform-to-Service-BridgeHealth-15e01fb05bf18092b67af29a49abfc9a

### CODE REVIEW

File: aws/eks-service/variables.tf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  description = "The zone id for adding hostnames for services"
+}
+
+variable "alb_dns" {
+  description = "The DNS of the ALB from K8s cluster"
+}
+
+variable "region" {
+  description = "Region for getting ALB hosted zone ID"
+}
```
Comment (spartan-ytruong) [inline aws/eks-service/variables.tf:28]: please add variable type

File: aws/eks-service/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,3 @@
+output "pod_role" {
+  value = aws_iam_role.this.arn
+}
```
Comment (spartan-ytruong) [inline aws/eks-service/outputs.tf:3]: please add description

File: aws/eks-service/variables.tf (hunk 1 above)
Comment (spartan-tainguyen) [reply aws/eks-service/variables.tf]: I forgot to create a commit for the last changes, thank you.
Updated

File: aws/eks-service/versions.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+terraform {
+  required_providers {
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.33"
+    }
+  }
+}
```
Comment (spartan-ytruong) [inline aws/eks-service/versions.tf:8]: lack of aws provider version constraint

File: aws/eks-service/main.tf [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            "StringEquals": {
+                "${var.eks_oidc_provider.url}:sub": "system:serviceaccount:${var.service.name}:${local.service_account_name}",
+                "${var.eks_oidc_provider.url}:aud": "sts.amazonaws.com"
+            }
+        }
+    }
+  ]
+}
+EOF
+}
```
Comment (spartan-ytruong) [inline aws/eks-service/main.tf:82]: we should use `data.aws_iam_policy_document` for this one

**Review by spartan-ytruong — APPROVED** [review]

**Pull request merged**